MiniApp/Classes/core/RealMiniApp.swift

```diff
@@ -253,27 +252,23 @@ internal class RealMiniApp {
                                               apiClient: self.miniAppClient) { (result) in
             switch result {
             case .success(let metaData):
-                if clearPermissions {
-                    self.cleanUpCustomPermissions(appId: appId, latestManifest: metaData)
-                }
                 completionHandler(.success(metaData))
             case .failure(let error):
                 completionHandler(.failure(error))
             }
         }
     }
 
-    /// In Preview mode, when Codebase is updated, & if there is a change in Manifest, it is good to delete the old Custom Permissions from the Keychain.
-    /// Because the user has to agree to the new MiniAppManifest, also this will help to compare the latest manifest.
-    func cleanUpCustomPermissions(appId: String, latestManifest: MiniAppManifest) {
-        if self.getCachedManifestData(appId: appId) != latestManifest {
-            miniAppPermissionStorage.removeKey(for: appId)
-        }
-    }
-
     /// Method to remove the unused/deleted items from the Keychain
     func cleanUpKeychain() {
-        self.miniAppStatus.removeUnusedCustomPermissions()
+        miniAppStatus.removeUnusedCustomPermissions()
+        miniAppStatus.removeManifestsFromKeychain()
+        if let currentVersion = MiniAppAnalytics.sdkVersion {
+            let defaults = UserDefaults(suiteName: MiniAppStatus.userDefaultsKey)
+            defaults?.set(currentVersion, forKey: MiniAppStatus.lastVersionKey)
+            defaults?.synchronize()
```

Comment:
stores the current version of the SDK in keychain, so next time the SDK is updated we can use this property to make some cleaning if necessary

Reply:
Good job 👍🏻